storops/unity/resource/lun.py

```diff
@@ -18,14 +18,163 @@
+
+        if sp is not None:
+            spNode = sp.to_node_enum()
+            NodeEnum.verify(spNode)
+
+        req_body = {}
+        if name is not None:
+            req_body['name'] = name
+        if description is not None:
+            req_body['description'] = description
+        if is_repl_dst is not None:
+            req_body['replicationParameters'] = {
+                    'isReplicationDestination': is_repl_dst
+            }
+
+        # TODO:iolimit_policy and snap_schedule
+        # Compose lun parameters
+        req_body['lunParameters'] = {}
+        if is_thin is not None:
+            req_body['lunParameters']['isThinEnabled'] = is_thin
+        if size is not None:
+            req_body['lunParameters']['size'] = size
+        if sp is not None:
+            req_body['lunParameters']['defaultNode'] = spNode
+        if tiering_policy is not None:
+            req_body['lunParameters']['fastVPParameters'] = {
+                    'tieringPolicy': tiering_policy
+                }
+
+        # compose host access parameters
+        if host_access is not None:
+            req_body['lunParameters']['hostAccess'] = []
+            for ha in host_access:
+                HostLUNAccessEnum.verify(ha['accessMask'])
+                req_body['lunParameters']['hostAccess'].append(
+                    {
+                        'host': ha['host'],
+                        'accessMask': ha['accessMask']
+                    }
+                )
+        # end if
+
+        resp = self._cli.action(UnityStorageResource().resource_class,
+                                self.get_id(), 'modifyLun', **req_body)
+        resp.raise_if_err()
+        return resp
+
+    def delete(self, async=False, force_snap_delete=False,
+               force_vvol_delete=False):
+        sr = self.storage_resource
+        if not self.existed or sr is None:
+            raise UnityResourceNotFoundError(
+                'cannot find lun {}.'.format(self.get_id()))
+        resp = self._cli.delete(sr.resource_class, self.get_id(),
+                                forceSnapDeletion=force_snap_delete,
+                                forceVvolDeletion=force_vvol_delete,
+                                async=async)
+        resp.raise_if_err()
+        return resp
+
+    def attach_to(self, host, max_retires=3,
+                  access_mask=HostLUNAccessEnum.PRODUCTION):
+        # TODO: max_retires to retry
+        host_access = [{'host': host, 'accessMask': access_mask}]
+        # If this lun has been attached to other host, don't overwrite it.
+        if self.host_access:
+            host_access += [{'host': item.host,
+                            'accessMask': item.access_mask} for item
+                            in self.host_access if host.id != item.host.id]
+
+        resp = self.modify(host_access=host_access)
+        return resp
+
+    def detach_from(self, host, max_retires=3):

```

Comment:
Same here.  Please remove the unused parameter.
If you really want to retry, use the `@retry` decorator.